bootloader/lk.go

```diff
@@ -0,0 +1,203 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package bootloader
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/bootloader/lkenv"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+type lk struct {
+	inRuntimeMode bool
+}
+
+// newLk create a new lk bootloader object
+func newLk() Bootloader {
+	e := &lk{}
+	if !osutil.FileExists(e.envFile()) {
+		return nil
+	}
+
+	// XXX: in the long run we want this to go away, we probably add
+	//      something like "boot.PrepareImage()" and add an (optional)
+	//      method "PrepareImage" to the bootloader interface that is
+	//      used to setup a bootloader from prepare-image if things
+	//      are very different from runtime vs image-building mode.
+	//
+	// determine mode we are in, runtime or image build
+	e.inRuntimeMode = (dirs.GlobalRootDir == "/")
```

Comment:
can't we use something different to detect runtime vs not?  if we keep this way we need some kind of image test because if we change how we GlobalRootDir there we won't notice

Reply:
it's also a bit confusing that we use envFile just above but it actually depends on this flag that we set here

Reply:
@mvo5  I'll do something intermediate in a separate PR, I think I wanted anyway to turn:

booloader.Find() ->  bootloader.Find(rootdir)  where rootdir = "" uses GlobalRootDir

I can easily change that a well into  booloader.Find(string, *Options), the we can a flag to here.

Reply:
I proposed https://github.com/snapcore/snapd/pull/7418, once that is merged, it can be merged back here and at the cost of some churn avoid the need for this kind of code

Reply:
Thanks @pedronis - I merged 7418 and updated the code here to use it.

Reply:
My commit does the minimal change - we could expand it further and use your new "Options" argument in Find() for LK. We probably also need to tweak InstallBootConfig so that ConfigFiles takes an argument as well or is just replaced with something else.

Reply:
yes, that was the point of the new argument. We don't need to tweak InstallBootConfig a lot, we know the different bootloaders there